binr/rabin2/rabin2.c

```diff
@@ -189,25 +196,27 @@ static bool extract_binobj (const RBinFile *bf, const RBinObject *o, int idx) {
 	free (outfile);
 	free (outpath);
 	free (path);
+	free (bytes);
+
 	return res;
 }
 
 static int rabin_extract(int all) {
-	RBinObject *obj = NULL;
+	RBinXtrData *data = NULL;
 	int res = false;
 	RBinFile *bf = r_bin_cur (bin);
 
 	if (!bf) return res;
 	if (all) {
 		int idx = 0;
 		RListIter *iter;
-		r_list_foreach (bf->objs, iter, obj) {
-			res = extract_binobj (bf, obj, idx++);
+		r_list_foreach (bf->xtr_data, iter, data) {
+			res = extract_binobj (bf, data, idx++);
```

Comment:
check res result and get out of the loop in case of error the same in the else clause